File: tweeter-z-frontend/tweeter/src/app/model/tweet-dto.ts
import { CommentDTO } from "./comment-dto";

export class TweetDTO {
    tweetId!: number;
    userId!: number;
    userHandle!: string;
    userName!: string;
    
    content!: string;
    createdAt!: Date;
    updatedAt!: Date;
    comments!: CommentDTO[];
    likes!: number[]; // User IDs who liked the tweet
    imageUrls!: string[];
    videoUrls!: string[];
    showComments: boolean = false; // Default value
  
    constructor(init?: Partial<TweetDTO>) {
      Object.assign(this, init);
    }
  }
  

File: tweeter-user-timeline-service/src/main/java/com/tweetapp/model/TweetDTO.java
package com.tweetapp.model;

import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.Date;
import java.util.List;

@Data
public class TweetDTO {
    private Long tweetId;

    private Long userId;
    private String userHandle;
    private String userName;

    private String content;

    private Date createdAt;
    private Date updatedAt;

    private String[] imageUrls;
    private String[] videoUrls;

    private List<CommentDTO> comments;
    private List<Long> likes; // User IDs of users who liked the tweet

    private boolean showComments = false;


}
